app/api/orders.py

```diff
@@ -456,3 +460,44 @@ def create_paypal_payment(order_identifier):
         return jsonify(status=True, payment_id=response)
     else:
         return jsonify(status=False, error=response)
+
+
+@alipay_blueprint.route('/create_source/<string:order_identifier>', methods=['GET', 'POST'])
+def create_source(order_identifier):
+    """
+    Create a source object for alipay payments.
+    :param order_identifier:
+    :return: The alipay redirection link.
+    """
+    try:
+        order = safe_query(db, Order, 'identifier', order_identifier, 'identifier')
+        source_object = AliPayPaymentsManager.create_source(amount=int(order.amount), currency='usd',
+                                                            redirect_return_uri=url_for('alipay_blueprint.alipay_return_uri',
+                                                            order_identifier=order.identifier, _external=True))
+        order.order_notes = source_object.id
+        save_to_db(order)
+        return jsonify(link=source_object.redirect['url'])
+    except TypeError:
+        return BadRequestError({'source': ''}, 'Source creation error').respond()
+
+
+@alipay_blueprint.route('/alipay_return_uri/<string:order_identifier>', methods=['GET', 'POST'])
+def alipay_return_uri(order_identifier):
+    """
+    Create a source object for alipay payments.
+    :param order_identifier:
+    :return: JSON response of the payment status.
+    """
+    try:
+        charge_response = AliPayPaymentsManager.charge_source(order_identifier)
+        print(charge_response)
+        if charge_response.status == 'succeeded':
+            # return make_frontend_url('/orders/{}/view'.format(order_identifier))
+            order = safe_query(db, Order, 'identifier', order_identifier, 'identifier')
+            order.status = 'completed'
+            save_to_db(order)
+            return redirect(make_frontend_url('/orders/{}/view'.format(order_identifier)))
+        else:
+            return jsonify(status=False, error='Charge object failure')
```

Comment:
@mrsaicharan1 can't this use standard exceptions/errors which are being used in the server so far?

Reply:
```
redirect(make_frontend_url('/orders/{}/view'.format(order_identifier)))
```

This statement is unreachable

Reply:
@iamareebjamal Can you please confirm why? I was able to go to the completed page via this statement.

Reply:
There's a return statement just before it

Reply:
@iamareebjamal Oh sorry, that was meant to be removed. That's why I commented it. Will remove it now.

Reply:
Also, please never redirect to a frontend URL. This is not open-event-frontend-server, this is open-event-server. This redirection will make it impossible to integrate it in Android Apps

Reply:
yes, I wanted to discuss about that. Would it be okay if I return `return jsonify(status=True, payment_id='xxxxx')`?

Reply:
If this is the way it is handled in other payment gateways as well, then let's merge it and work on fixing this behavior

Reply:
@mrsaicharan1 Yes, please return success JSON and handle it on frontend, i.e., the frontend itself with route to the correct page

Reply:
> If this is the way it is handled in other payment gateways as well, then let's merge it and work on fixing this behavior

This would be better because this PR has too many changes. It would be better If you could merge this. Even Paypal requires android support, as stated here https://github.com/fossasia/open-event-server/issues/5961. So will fix both of them in a follow up  PR.

Reply:
@iamareebjamal Will fix the remaining hound issue, remove some redundant print statements and get back to you

Reply:
@iamareebjamal Please check.